perun-core/src/main/java/cz/metacentrum/perun/core/impl/VosManagerImpl.java

```diff
@@ -358,5 +384,125 @@ public int getVosCount(PerunSession sess) {
 		}
 	}
 
+	@Override
+	public BanOnVo setBan(PerunSession sess, BanOnVo banOnVo) {
+		Utils.notNull(banOnVo.getValidityTo(), "banOnVo.getValidityTo");
+		try {
+			int newId = Utils.getNewId(jdbc, "vos_bans_id_seq");
+
+			jdbc.update("insert into vos_bans(" +
+							"id, " +
+							"description, " +
+							"banned_to, " +
+							"member_id, " +
+							"vo_id, " +
+							"modified_by, " +
+							"created_by, " +
+							"created_by_uid, " +
+							"modified_by_uid) values (?,?,?,?,?,?,?,?,?)",
+					newId,
+					banOnVo.getDescription(),
+					Compatibility.getDate(banOnVo.getValidityTo().getTime()),
+					banOnVo.getMemberId(),
+					banOnVo.getVoId(),
+					sess.getPerunPrincipal().getActor(),
+					sess.getPerunPrincipal().getActor(),
+					sess.getPerunPrincipal().getUserId(),
+					sess.getPerunPrincipal().getUserId()
+			);
+
+			banOnVo.setId(newId);
+
+			return banOnVo;
+		} catch(RuntimeException ex) {
+			throw new InternalErrorException(ex);
+		}
+	}
+
+	@Override
+	public BanOnVo getBanById(PerunSession sess, int banId) throws BanNotExistsException {
+		try {
+			return jdbc.queryForObject("select " + banOnVoMappingSelectQuery + " from vos_bans where id=? ",
+					BAN_ON_VO_MAPPER, banId);
+		} catch (EmptyResultDataAccessException ex) {
+			throw new BanNotExistsException("Ban with id " + banId + " not exists for any vo.");
+		} catch (RuntimeException ex) {
+			throw new InternalErrorException(ex);
+		}
+	}
+
+	@Override
+	public List<BanOnVo> getBansForVo(PerunSession sess, int voId) {
+		try {
+			return jdbc.query("select " + banOnVoMappingSelectQuery + " from vos_bans where vo_id=?",
+					BAN_ON_VO_MAPPER, voId);
+		} catch (EmptyResultDataAccessException ex) {
```

Comment:
Query does not throw EmptyResultDataAccessException.

Reply:
Removed catch.